javalib/src/main/scala/java/util/concurrent/atomic/Striped64.scala

```diff
@@ -0,0 +1,288 @@
+// Ported from JSR 166 revision 1.28
+
+package java.util.concurrent.atomic
+
+import java.lang.Double._
+import java.util.Arrays;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.DoubleBinaryOperator;
+import java.util.function.LongBinaryOperator;
+import scala.scalanative.annotation._
+import scala.scalanative.unsafe._
+import scala.scalanative.libc.atomic.{CAtomicInt, CAtomicLongLong, memory_order}
+import scala.scalanative.runtime.{fromRawPtr, Intrinsics}
+
+/*
+ * Written by Doug Lea with assistance from members of JCP JSR-166
+ * Expert Group and released to the public domain, as explained at
+ * http://creativecommons.org/publicdomain/zero/1.0/
+ */
+
+@SuppressWarnings(Array("serial"))
+object Striped64 {
+
+  final private[atomic] class Cell private[atomic] (var value: Long) {
+    @volatile private[concurrent] var _value = value
+
+    @alwaysinline def valueAtomic() = new CAtomicLongLong(
+      fromRawPtr(Intrinsics.classFieldRawPtr(this, "_value"))
+    )
+
+    final private[atomic] def cas(cmp: Long, `val`: Long) =
+      valueAtomic().compareExchangeWeak(
+        cmp,
+        `val`,
+        memory_order.memory_order_release
+      )
```

Comment:
Dumb question: in JSR166 I see the following comment, is that what you've done here with `memory_order_release`?

```
     * JVM intrinsics note: It would be possible to use a release-only
     * form of CAS here, if it were provided.
```

Reply:
Ah, sorry I am looking at an old version. This matches the latest code :)

Reply:
yeah I refered to [VarHandle.weakCompareAndSetRelease](https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/lang/invoke/VarHandle.html#weakCompareAndSetRelease(java.lang.Object...)), and implemented it for scala-native.